process.go

```diff
@@ -0,0 +1,159 @@
+	//       the original informational blurb has been copied from...
+	//
+
+	// Because containers are spawned in a two step process you will need
+	// a binary that will be executed as the init process for the
+	// container. In libcontainer, we use the current binary,
+	// in /proc/self/exe, to be executed as the init process, and use arg
+	// "init", we call the first step process "bootstrap", so you always
+	// need a "init" function as the entry of "bootstrap"
+	//
+	if len(os.Args) > 1 && os.Args[1] == "init" {
+		runtime.GOMAXPROCS(1)
+		runtime.LockOSThread()
+		factory, _ := libcontainer.New("")
+		if err := factory.StartInitialization(); err != nil {
+			stdlog("process.init() --> failure to initialize")
+			stdlog(err.Error())
+			return
+		}
+		panic("--this line should have never been executed, congratulations--")
+	}
+}
+
+//! Function to start a process, as per the global rootfs.
+/*
+ * @param    Job    given job
+ *
+ * @return   Process    newly generated process
+ * @return   error      error message, if any
+ */
+func StartProcess(j Job) (p *Process, err error) {
+
+	// Spawn a containerizer.
+	containerizer, err := libcontainer.New(rootfs, libcontainer.Cgroupfs)
+
+	// check if an error occurred
+	if err != nil {
+		return nil, err
+	}
+
+	// Make a crypto safe pseudo random string as the id.
+	containerID := spawnPseudorandomString(16)
+
+	// Set the location of the container.
+	containerName := filepath.Base(j.Path) + "_" + containerID
+
+	// Generate a new libcontainer config.
+	config := generateLibcontainerConfig(containerName, j.Hostname)
+
+	// Use the containerizer to make a container.
+	container, err := containerizer.Create(containerID, config)
+
+	// check if any error occurred
+	if err != nil {
+		return nil, err
+	}
+
+	// Assemble a new process.
+	process := &libcontainer.Process{
+		Args:   append([]string{j.Path}, j.Args...),
+		Env:    j.Env,
+		Stdin:  os.Stdin,
+		Stdout: os.Stdout,
+		Stderr: os.Stderr,
+	}
+
+	// get the container to start the given process
+	err = container.Start(process)
+
+	// safety check, make sure this actually worked; also clean mem in the
+	// event that this is no longer able to continue
+	if err != nil {
+		container.Destroy()
+		return nil, err
+	}
+
+	// Assemble a reference to the newly generated process.
+	return &Process{uuid: j.Pid, container: container, proc: process}, nil
+}
+
+//! Function to stop a process, as per the global rootfs.
+/*
+ * @param    Process    given process to stop
+ *
+ * @return   error      error message, if any
+ */
+func StopProcess(p *Process) error {
+
+	// input validation
+	if p == nil {
+		return errorf("StopProcess() --> invalid input\n")
+	}
+
+	// safety check, ensure this is actually a process
+	if p.proc == nil {
+		debugf("StopProcess() --> no such process, ignoring")
+		return nil
+	}
+
+	// Have the OS send the kill signal to the process.
+	err := p.proc.Signal(os.Kill)
```

Comment:
I suppose this could send an interrupt first.

https://github.com/rbisewski/lclusterd/commit/0950acd7dcd914f9d96bc83da81170dfc13cefdc

Maybe a better way might have been to wait a few seconds and then signal release or signal kill. Your thoughts on the ideal way to handle this?